### ISSUE TITLE
runtime error: slice bounds out of range

### ISSUE THREAD

**Issue by benubois** [wal-g/wal-g#19]
Hello,

I'm interested in trying out wal-g. Currently I'm trying to restore an existing backup that was created by *wal-e* and I'm getting an error:

```
envdir /etc/wal-e.d/reader /usr/local/bin/wal-g backup-fetch /var/lib/postgresql/9.2/main LATEST
BUCKET: database-backups
SERVER: path/to/backup
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
main.main()
	/home/travis/gopath/src/github.com/wal-g/wal-g/cmd/wal-g/main.go:136 +0x1890
```

These are the environment variables in `/etc/wal-e.d/reader/`:

```
ls -1 /etc/wal-e.d/reader/
AWS_ACCESS_KEY_ID
AWS_REGION
AWS_SECRET_ACCESS_KEY
WALE_S3_PREFIX
```

Any idea what I'm doing wrong?

Thanks!

**Comment by benubois** [comment]
Appears the problem is related to using `LATEST`. Specifying a base backup seems to work, i.e.

```
envdir /etc/wal-e.d/reader /usr/local/bin/wal-g backup-fetch /var/lib/postgresql/9.2/main base_XYZ
```

**Comment by fdr** [comment]
Thanks for narrowing things down, perhaps you might consider writing a patch and a test?

**Comment by reyalcantara** [comment]
I am having issue both for 
**backup-push:**
```
$ envdir /etc/wal-g.d/env /usr/local/bin/wal-g backup-push /var/lib/postgresql/9.6/main/
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/wal-g/wal-g.Configure(0x10, 0xc4201784e0, 0x0, 0x18)
	/home/travis/gopath/src/github.com/wal-g/wal-g/upload.go:77 +0xa6e
main.main()
	/home/travis/gopath/src/github.com/wal-g/wal-g/cmd/wal-g/main.go:83 +0x1f6
```
**wal-push:**
```
2017-09-21 15:57:17.057 UTC [14273] DETAIL:  The failed archive command was: envdir /etc/wal-g.d/env /usr/local/bin/wal-g wal-push pg_xlog/000000010000000000000001
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/wal-g/wal-g.Configure(0x10, 0xc420176500, 0x0, 0x18)
	/home/travis/gopath/src/github.com/wal-g/wal-g/upload.go:77 +0xa6e
main.main()
	/home/travis/gopath/src/github.com/wal-g/wal-g/cmd/wal-g/main.go:83 +0x1f6
2017-09-21 15:57:18.072 UTC [14273] LOG:  archive command failed with exit code 2
2017-09-21 15:57:18.072 UTC [14273] DETAIL:  The failed archive command was: envdir /etc/wal-g.d/env /usr/local/bin/wal-g wal-push pg_xlog/000000010000000000000001
2017-09-21 15:57:18.072 UTC [14273] WARNING:  archiving transaction log file "000000010000000000000001" failed too many times, will try again later
```

**Comment by patf** [comment]
I ran into this with `backup-fetch`. It seems to have been caused by a trailing slash in my `WALE_S3_PREFIX` value. Removing it fixed `backup-fetch` both with `LATEST` and a backup name.

**Comment by x4m** [comment]
For LATEST will be fixed in #33, probably, today.
For things reported by @reyalcantara seems like it's a bug, needs some further investigation.

**Comment by x4m** [comment]
Fixed.
